### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by Richa-rani02**
Implemented Below features

1. Login with validation
2. signup with validation and placeholder for error message
3. logout
4. store userinfo and token in localstorage
5. show username on header

### CODE REVIEW

File: src/components/Header.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
-import { FaUserCircle } from "react-icons/fa";
-import {AiOutlineLogout} from "react-icons/ai";
+import {useAuth} from "../context/auth-context";
```
Comment (VedanthB) [inline src/components/Header.jsx:1]: you can do a barrel import/export to make code look more clean

**Review by VedanthB — APPROVED** [review]
Good work Richa,

things i liked 
- clean organised code
- good variable naming 

You can merge the PR 🚀

File: src/pages/Authentication/Auth.jsx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import { Header } from "../../components";
+import {Login} from "./Login";
+import { Signup } from "./Signup";
+import { useLocation } from "react-router-dom";
+export const Auth = () => {
+    const location=useLocation();
+    console.log(location.pathname);
```
Comment (prathmeshjagtap) [inline src/pages/Authentication/Auth.jsx:7]: you can remove console.log

**Review by prathmeshjagtap — APPROVED** [review]
Code is Clean 
Also, Good Naming Conventions

**Pull request merged**